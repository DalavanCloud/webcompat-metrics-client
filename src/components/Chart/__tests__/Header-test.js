/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";

import { Header } from "../";

it("renders Header default correctly", () => {
  const tree = renderer
    .create(
      <Header title="I'm a title" style={{ color: "green" }}>
        <div>{"Header"}</div>
      </Header>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Header without style", () => {
  const tree = renderer
    .create(
      <Header title="I'm a title">
        <span>{"Header"}</span>
      </Header>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Header without children", () => {
  const tree = renderer.create(<Header title="I'm a title" />).toJSON();
  expect(tree).toMatchSnapshot();
});
