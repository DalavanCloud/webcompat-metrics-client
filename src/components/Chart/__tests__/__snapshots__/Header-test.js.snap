// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header default correctly 1`] = `
<header
  className="header"
  style={
    Object {
      "color": "green",
    }
  }
>
  <section>
    <h3
      className="headerTitle"
    >
      I'm a title
    </h3>
  </section>
  <section>
    <div>
      Header
    </div>
  </section>
</header>
`;

exports[`renders Header without children 1`] = `
<header
  className="header"
  style={Object {}}
>
  <section>
    <h3
      className="headerTitle"
    >
      I'm a title
    </h3>
  </section>
</header>
`;

exports[`renders Header without style 1`] = `
<header
  className="header"
  style={Object {}}
>
  <section>
    <h3
      className="headerTitle"
    >
      I'm a title
    </h3>
  </section>
  <section>
    <span>
      Header
    </span>
  </section>
</header>
`;
